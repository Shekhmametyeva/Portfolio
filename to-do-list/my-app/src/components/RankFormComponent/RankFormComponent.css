.form__button {
    height: 40px;
    padding: 0 5px;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 5px;
    border: 2px solid #9b9a9a;
    font-size: 1.1rem;
    cursor: pointer;
    border-radius: 5px;
    box-shadow: 3px 3px 3px #016801;
    transition: all 0.2s linear;
}
.form__button:hover {
    background-color: #01680170;
}
.form__button.active {
    background-color: #01680170;
    border: 2px solid #016801;
    box-shadow: none;
}