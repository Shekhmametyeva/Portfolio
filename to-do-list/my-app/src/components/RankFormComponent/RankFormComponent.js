import ButtonSvgComponent from '../ButtonSvgComponent/ButtonSvgComponent';
import './RankFormComponent.css';

function RankFormComponent (props) {
    
    const button = props.rank.map(el => {
        return (
            <div 
                key={el} 
                className={props.activeRank === el ? 'form__button active' : 'form__button'} 
                value={el}
                onClick={() => props.changeActiveRanc(el)}>
                    <ButtonSvgComponent type='button' name={el} class='rank__button'/>
                    <p>{el}</p>
                
            </div>)
    })
    return (
        <div className='form__button__container'>
            {button}
        </div>
    )
}

export default RankFormComponent