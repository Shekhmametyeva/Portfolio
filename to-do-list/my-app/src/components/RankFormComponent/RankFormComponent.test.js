import {render, screen} from '@testing-library/react';
import RankFormComponent from './RankFormComponent';

describe('тестируем FormComponent', () => {
    const ranksList = ['Личные', 'Семья', 'Работа', 'Покупки', 'Учеба'];
    test('render RankFormComponent', () => {
        render(<RankFormComponent ranksList={ranksList} />);
        const btns = screen.getAllByRole('button');
        expect(btns.length).toBe(ranksList.length);
        expect(btns).toMatchSnapshot()
    });
    test('render RankFormComponent c классом active', () => {
        render(<RankFormComponent ranksList={ranksList} activeRank='Семья'/>);
        const btnRank = screen.getAllByTestId('rank-button');
        expect(btnRank[1]).toHaveClass('active')
    });
})