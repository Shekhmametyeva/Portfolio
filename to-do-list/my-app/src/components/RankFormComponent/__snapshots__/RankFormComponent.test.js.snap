// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестируем FormComponent render RankFormComponent 1`] = `
Array [
  <button
    class="rank__button button"
    type="button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#Личные"
      />
    </svg>
  </button>,
  <button
    class="rank__button button"
    type="button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#Семья"
      />
    </svg>
  </button>,
  <button
    class="rank__button button"
    type="button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#Работа"
      />
    </svg>
  </button>,
  <button
    class="rank__button button"
    type="button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#Покупки"
      />
    </svg>
  </button>,
  <button
    class="rank__button button"
    type="button"
  >
    <svg
      class="icon"
    >
      <use
        xlink:href="sprite.svg#Учеба"
      />
    </svg>
  </button>,
]
`;
